src/frontends/tensorflow_common/src/helper_transforms/tensor_list_ops_resolver.cpp

```diff
@@ -209,3 +329,304 @@ ov::frontend::tensorflow::pass::TensorListGetItemReplacer::TensorListGetItemRepl
+
+        const auto& body = loop->get_function();
+        const auto& body_params = body->get_parameters();
+        const auto& body_results = body->get_results();
+        const auto& special_body_ports = loop->get_special_body_ports();
+        const auto& input_descriptions = loop->get_input_descriptions();
+        const auto& output_descriptions = loop->get_output_descriptions();
+
+        if (!condition_matcher.match(body_results[special_body_ports.body_condition_output_idx]->output(0))) {
+            return false;
+        }
+
+        const auto& condition_map = condition_matcher.get_pattern_value_map();
+        int64_t counter_step = -1;
+        if (!get_constant_value(condition_map.at(counter_step_label).get_node_shared_ptr(), counter_step) ||
+            counter_step != 1) {
+            return false;
+        }
+
+        // get initial value of counter
+        int64_t initial_counter = 0;
+        auto counter_param = ov::as_type_ptr<v0::Parameter>(condition_map.at(counter_label).get_node_shared_ptr());
+        if (!counter_param) {
+            return false;
+        }
+
+        for (const auto& input_desc : loop->get_input_descriptions()) {
+            auto body_param_idx = input_desc->m_body_parameter_index;
+            auto input_idx = input_desc->m_input_index;
+            if (body_params[body_param_idx] != counter_param) {
+                continue;
+            }
+
+            // it must be merged input and incremented each iteration
+            auto merged_desc = ov::as_type_ptr<MergedD>(input_desc);
+            if (!merged_desc) {
+                return false;
+            }
+            auto result_idx = merged_desc->m_body_value_index;
+            auto update_counter = body_results[result_idx]->input_value(0).get_node_shared_ptr();
+            if (update_counter != condition_map.at(updated_counter_label).get_node_shared_ptr()) {
+                return false;
+            }
+
+            // get initial value of counter
+            if (!get_constant_value(loop->get_input_node_shared_ptr(input_idx), initial_counter)) {
+                return false;
+            }
+
+            // suitable counter-parameter is found and checked
+            break;
+        }
+
+        if (initial_counter != 0) {
+            return false;
+        }
+
+        // collect vector of updated Parameter indices (they will be converted to SlicedInput parameters)
+        // and Result nodes (they will be converted to ConcatOutput results)
+        // also, some parameters and results can be removed and they are directly connected
+        // to updated Parameter/Result nodes
+        std::vector<uint64_t> remove_parameter_ids;
+        std::vector<uint64_t> update_parameter_ids, update_result_ids;
+        for (const auto& target_input : counter_param->get_output_target_inputs(0)) {
+            update_parameter_to_slice_input(target_input.get_node()->shared_from_this(),
```

Comment:
yes, we should. This transformation change input port only for those which are suitable. The others remain the same. It is generic transformations, oriented not only for RNN. It can transform multiple inputs in Loop to SlicedInput.